pkg/network/group.go

```diff
@@ -108,12 +109,7 @@ func (x *AddressGroup) FromIterator(iter MultiAddressIterator) error {
 		return errors.New("missing network addresses")
 	}
 
-	if cap(as) >= addrNum {
-		as = as[:0]
-	} else {
-		as = make(AddressGroup, 0, addrNum)
-	}
-
+	as = slices.Grow(as, addrNum)[:0]
```

Comment:
I think it should be `slices.Grow(as[:0], addrNum)` then. But it's an interesting pattern as well since it allows to handle `nil`.

Reply:
so what will we do here? it was not a mistake, in my head it is "grow slice capacity, make its length zero". do not say the inverted version is bad just wanted to do a different thing

Reply:
Well, they're the same in effect since `Grow` doesn't change length. Whatever you prefer.